packages/main/src/Calendar.ts

```diff
@@ -423,21 +469,60 @@ class Calendar extends CalendarPart {
 		});
 
 		this.timestamp = timestamp;
-		const defaultPrevented = !this.fireEvent<CalendarChangeEventDetail>("selected-dates-change", { timestamp, dates: [...selectedDates], values: datesValues }, true);
+		const defaultPrevented = !this.fireEvent<CalendarChangeEventDetail>(
+			"selected-dates-change",
+			{ timestamp, dates: [...selectedDates], values: datesValues },
+			true,
+		);
 		if (!defaultPrevented) {
 			this._setSelectedDates(selectedDates);
 		}
 	}
 
 	onSelectedMonthChange(e: CustomEvent<MonthPickerChangeEventDetail>) {
 		this.timestamp = e.detail.timestamp;
-		this._currentPicker = "day";
+		if (this.pickerModes.includes("day")) {
+			this._currentPicker = "day";
+		} else {
+			const selectedDates = [this.timestamp * 1000];
```

Comment:
Why do you multiple by 1000 here? Wouldn't that change the format of the `selectedDates` property in the event detail object? Before, users were receiving one type of number, and now they will be receiving another type, which will cause bugs.